### ISSUE TITLE
keymap inheritance

### ISSUE THREAD

**Issue by Dagon** [UltimateHackingKeyboard/agent#1206]
I have two UHKs (for work and for home), and will likely buy another for travel.  I have 3 keymaps I switch between on them, which only vary by a small amount, and another 2 keymaps I use for backups when I'm experimenting with things.

It would be REALLY nice to be able to specify a base keymap, and an inheritance hierarchy of derived keymaps.  A derived keymap would have two types of "NONE" mapping for keys.  The normal none would make the key non-functional as it does today.  The "inherit" mapping would make it behave in whatever way the parent keymap is defined.  

Importantly, a change in the parent keymap would apply to all the keymaps derived, except overridden mappings on those derived keymaps.

I think this would be mostly an Agent change, to calculate the individual keymaps to send to the keyboard for operations.  It could be a firmware change if it saves memory or something, or is easier to implement that way.

**Comment by mondalaci** [comment]
Back when I designed Agent and the UHK firmware, I was thinking about this feature. I eventually concluded that while it'd useful for some, it would add to the complexity far too much compared to its value. It'd also expose additional complexity for everyone. I think you are the first or possibly the second person recommending it, so it's certainly a niche feature.

I'm sorry, but I cannot justify implementing this feature given the above. I hope you understand it, and thank you very much for your great support!
